lib/activerecord-import/adapters/postgresql_adapter.rb

```diff
@@ -24,12 +27,113 @@ def next_value_for_sequence(sequence_name)
 
   def post_sql_statements( table_name, options ) # :nodoc:
     unless options[:primary_key].blank?
-      super(table_name, options) << (" RETURNING #{options[:primary_key]}")
+      super(table_name, options) << ("RETURNING #{options[:primary_key]}")
     else
       super(table_name, options)
     end
   end
 
+  # Add a column to be updated on duplicate key update
+  def add_column_for_on_duplicate_key_update( column, options={} ) # :nodoc:
+    arg = options[:on_duplicate_key_update]
+    if arg.is_a?( Hash )
+      columns = arg.fetch( :columns ) { arg[:columns] = [] }
+      case columns
+      when Array then columns << column.to_sym unless columns.include?( column.to_sym )
+      when Hash then columns[column.to_sym] = column.to_sym
+      end
+    elsif arg.is_a?( Array )
+      arg << column.to_sym unless arg.include?( column.to_sym )
+    end
+  end
+
+  # Returns a generated ON CONFLICT DO NOTHING statement given the passed
+  # in +args+.
+  def sql_for_on_duplicate_key_ignore( table_name, *args ) # :nodoc:
+    arg = args.first
+    if arg.is_a?( Hash )
+      conflict_target = sql_for_conflict_target( arg ) if arg.is_a?( Hash )
+    end
+    " ON CONFLICT #{conflict_target}DO NOTHING"
```

Comment:
`sql_for_conflict_target` returns a string with a blank space at the end, so I think it will be ok.